Framework/Algorithms/src/ReflectometryWorkflowBase2.cpp

```diff
@@ -685,5 +692,90 @@ ReflectometryWorkflowBase2::getRunNumber(MatrixWorkspace const &ws) const {
   return "";
 }
 
+std::string
+ReflectometryWorkflowBase2::convertProcessingInstructionsToWorkspaceIndexes(
+    const std::string &instructions, MatrixWorkspace_const_sptr ws) const {
+  std::string converted = "";
+  std::string currentNumber = "";
+  std::string ignoreThese = "-,:+";
+  for (auto i = 0u; i < instructions.size(); ++i) {
+    if (std::find(ignoreThese.begin(), ignoreThese.end(), instructions[i]) !=
+        ignoreThese.end()) {
+      // Found a spacer so add currentNumber to converted after seperator
+      converted.append(workspaceIndexesToSpecNum(currentNumber, ws));
+      converted.push_back(instructions[i]);
+      currentNumber = "";
+    } else {
+      currentNumber.push_back(instructions[i]);
+    }
+  }
+  // Add currentNumber onto converted
+  converted.append(workspaceIndexesToSpecNum(currentNumber, ws));
+  return converted;
+}
+
+std::string ReflectometryWorkflowBase2::workspaceIndexesToSpecNum(
+    const std::string &num, MatrixWorkspace_const_sptr ws) const {
+  if (num == "0") {
```

Comment:
The error message here is misleading because the input is a spectrum number rather than a workspace index. I don't actually think we should have a special case for `0` - to be more generic, it should be handled in the same way as any other invalid spectrum number.

I think we can just remove this check. It will then report `Could not find spectrum number in any spectrum` which will be more consistent with the error you get for any other invalid spectrum number.